batch/batch/server/server.py

```diff
@@ -299,6 +308,7 @@ def _has_next_task(self):
         self._task_idx += 1
         self._current_task = self._tasks[self._task_idx] if self._task_idx < len(self._tasks) else None
 
+        await _delete_pod_by_name(self._pod_name)
```

Comment:
The line below should prob just be `self._pod_name = None` without the if statement.

Reply:
Should this be after we update the database below (update_with_log_ec)?